src/mir/optimizations.md

```diff
@@ -1 +1,45 @@
 # MIR optimizations
+
+MIR optimizations are optimizations run on the [MIR][mir] to produce better MIR
+before codegen. This is important for two reasons: first, it make the final
+generated executable code better, and second, it means that LLVM has less work
+to do, so compilation is faster.
```

Comment:
This part is particularly effective, because we can optimize generic functions, so all the monomorphisations are cheaper

Reply:
Oh, that's cool. I added a note.